nifi-nar-bundles/nifi-standard-services/nifi-oauth2-provider-bundle/nifi-oauth2-provider-service/src/main/java/org/apache/nifi/oauth2/StandardOauth2AccessTokenProvider.java

```diff
@@ -181,6 +205,7 @@ public class StandardOauth2AccessTokenProvider extends AbstractControllerService
     private volatile String authorizationServerUrl;
     private volatile OkHttpClient httpClient;
 
+    private volatile String clientAuthenticationStrategy;
```

Comment:
Changing the `AllowableValue` to a specific `enum` named `ClientAuthenticationStrategy` will allow this property to be more strongly typed.

Reply:
Done